test/ranges.jl

```diff
@@ -1324,6 +1333,7 @@ Base.:+(x, y::NotReal) = x + y.val
 Base.zero(y::NotReal) = zero(y.val)
 Base.rem(x, y::NotReal) = rem(x, y.val)
 Base.isless(x, y::NotReal) = isless(x, y.val)
+Base.isinteger(y::NotReal) = isinteger(y.val)
```

Comment:
Is this still necessary or a leftover? Without it, what happens?

Reply:
Without this, the test at the following line fails, since `steprange_last` checks whether `NotReal(1)` is an integer.
https://github.com/JuliaLang/julia/blob/e959dde5bbdc2a6a83b2524a9ebd8599c21c34c1/test/ranges.jl#L1337

Reply:
I worry about adding an API requirement we didn't have before. What about checking `isinteger(start+step)`?

Reply:
Sounds good to me.